test/adapter/json_api/parse_test.rb

```diff
@@ -130,6 +130,25 @@ def test_polymorphic
             }
             assert_equal(expected, parsed_hash)
           end
+
+          def test_sti_type
+            parsed_hash = ActiveModelSerializers::Adapter::JsonApi::Deserialization.parse!(@hash, sti_type_key: :_type)
+            assert_equal(@expected.merge(_type: 'Photo'), parsed_hash)
+
+            with_config(sti_type_key: :_type) do
+              parsed_hash = ActiveModelSerializers::Adapter::JsonApi::Deserialization.parse!(@hash)
+              assert_equal(@expected.merge(_type: 'Photo'), parsed_hash)
+            end
+          end
+
+          def test_namespaced_sti_type
+            hash = @hash
+            hash['data']['type'] = 'content--user-photos'
```

Comment:
Disclaimer: I'm using Mongoid, but the behavior seems to be quite the same than ActiveRecord except Mongoid uses the `:_type` attribute, which is why I allowed customization.

This type field deserialization is necessary to create STI objects from the API.

When creating object, the JSON payload has the same format than serialized output, including the `type` root data key: http://jsonapi.org/format/#crud-creating

Why would you add a `data['attributes']['type']` when it's already supposed to be set in `data['type']` ?